### PR TITLE
Updated Generator Notebooks

### PR DESCRIPTION
**Pull request by forman**
Updated Generator NBs with respect to #514.

Checklist:

* [x] Add unit tests and/or doctests in docstrings
* [x] Add docstrings and API docs for any new/modified user-facing classes and functions
* [x] AppVeyor CI passes
* [x] Test coverage remains or increases (target 100%)

Remember to close associated issues after merge!

### CODE REVIEW

**Review by TonioF — CHANGES_REQUESTED** [review]
Three of these notebooks fail. #1 and #2 do not have an 'edc-service-yml', for #4 get_cube_info fails with DataStoreError: Invalid parameterization detected: 'client_id' is a required property .

**Review by AliceBalfanz — CHANGES_REQUESTED** [review]
I agree with Tonio, three NBs fail due to missing edc-service.yml (NB nr. 1, 3 and 4.).
@TonioF, you must export your SH client ID and secrets before executing the NBs, then you don't get the error message about client_id.

**Review by pont-us — CHANGES_REQUESTED** [review]
Like the others, I of course get the same errors from the missing service configuration, but everything works fine when I copy `edc-service-dev-config.yml` in as `edc-service.yml` (and set the environment variables for SH access). I'm not sure what the preferred solution is here -- obviously we can't commit secrets to a public repo, but maybe it would be good to have a prominent notice at the start of the notebook that you need to obtain these credentials, and SH credentials, to make it work.

(Incidentally, I also got a lot of xarray "Failed to open Zarr store with consolidated metadata, falling back to try reading non-consolidated metadata." warnings from both xcube and xcube-sh in Notebook 4, but this is something that needs fixing in the code rather than the notebook, so not a job for this PR.)

**Comment by forman** [comment]
@AliceBalfanz @TonioF @dzelge @pont-us 

I've added the missing config file `edc-service.yml`. It now expects you to set some env vars:

```yaml
endpoint_url: "${EDC_ENDPOINT_URL}"
client_id: "${EDC_CLIENT_ID}"
client_secret: "${EDC_CLIENT_SECRET}"
```

File: examples/notebooks/generators/edc-service.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+endpoint_url: "${EDC_ENDPOINT_URL}"
```
Comment (pont-us) [inline examples/notebooks/generators/edc-service.yml:1]:
```suggestion
# This file specifies the credentials for the remote generator service used
# in the example notebooks. The settings below read these credentials from
# environment variables. To run the generator examples in the notebooks,
# either set these environment variables before starting the notebook, or
# edit this file to specify the credentials directly.

endpoint_url: "${EDC_ENDPOINT_URL}"
```

**Review by pont-us — CHANGES_REQUESTED** [review]
Looks good! My only requested change is the comment I've suggested adding to the `edc-service.yml` file.

**Comment by AliceBalfanz** [comment]
In 3_remote.ipynb in cell 8:

I used the dev cluster URL and credentials 

```
result = generator.get_cube_info(request)
result
```

```
---------------------------------------------------------------------------
ValidationError                           Traceback (most recent call last)
~/Desktop/projects/xcube-dev/xcube/core/gen2/remote/generator.py in _parse_response(self, response, response_type, request_data)
    239         try:
--> 240             return response_type.from_dict(response_data)
    241         except Exception as e:

~/Desktop/projects/xcube-dev/xcube/util/jsonschema.py in from_dict(cls, value)
    608         """Create instance from JSON-serializable dictionary *value*."""
--> 609         return cls.get_schema().from_instance(value)
    610 

~/Desktop/projects/xcube-dev/xcube/util/jsonschema.py in from_instance(self, instance)
    118         # TODO: support anyOf
--> 119         self.validate_instance(instance)
    120         return self._from_validated_instance(instance)

~/Desktop/projects/xcube-dev/xcube/util/jsonschema.py in validate_instance(self, instance)
    103         # date-time format, strict_rfc3339 or rfc3339-validator is required.
--> 104         jsonschema.validate(instance=instance, schema=self.to_dict(),
    105                             format_checker=jsonschema.draft7_format_checker,

~/miniconda3/envs/xcube-dev/lib/python3.9/site-packages/jsonschema/validators.py in validate(instance, schema, cls, *args, **kwargs)
    933     if error is not None:
--> 934         raise error
    935 

ValidationError: 'dataset_descriptor' is a required property

Failed validating 'required' in schema:
    {'additionalProperties': True,
     'properties': {'cost_estimation': {'additionalProperties': False,
                                        'properties': {'available': {'minimum': 0,
                                                                     'type': 'integer'},
                                                       'limit': {'minimum': 0,
                                                                 'type': 'integer'},
                                                       'required': {'minimum': 0,
                                                                    'type': 'integer'}},
                                        'required': ['required'],
                                        'type': 'object'},
                    'dataset_descriptor': {'additionalProperties': False,
                                           'properties': {'attrs': {'additionalProperties': True,
                                                                    'type': 'object'},
                                                          'bbox': {'items': [{'type': 'number'},
                                                                             {'type': 'number'},
                                                                             {'type': 'number'},
                                                                             {'type': 'number'}],
                                                                   'type': 'array'},
                                                          'coords': {'additionalProperties': {'additionalProperties': False,
                                                                                              'properties': {'attrs': {'additionalProperties': True,
                                                                                                                       'type': 'object'},
                                                                                                             'chunks': {'items': {'minimum': 0,
                                                                                                                                  'type': 'integer'},
                                                                                                                        'type': 'array'},
                                                                                                             'dims': {'items': {'minLength': 1,
                                                                                                                                'type': 'string'},
                                                                                                                      'type': 'array'},
                                                                                                             'dtype': {'minLength': 1,
                                                                                                                       'type': 'string'},
                                                                                                             'name': {'minLength': 1,
                                                                                                                      'type': 'string'}},
                                                                                              'required': ['dtype',
                                                                                                           'dims',
                                                                                                           'name'],
                                                                                              'type': 'object'},
                                                                     'type': 'object'},
                                                          'crs': {'minLength': 1,
                                                                  'type': 'string'},
                                                          'data_id': {'minLength': 1,
                                                                      'type': 'string'},
                                                          'data_type': {'minLength': 1,
                                                                        'type': 'string'},
                                                          'data_vars': {'additionalProperties': {'additionalProperties': False,
                                                                                                 'properties': {'attrs': {'additionalProperties': True,
                                                                                                                          'type': 'object'},
                                                                                                                'chunks': {'items': {'minimum': 0,
                                                                                                                                     'type': 'integer'},
                                                                                                                           'type': 'array'},
                                                                                                                'dims': {'items': {'minLength': 1,
                                                                                                                                   'type': 'string'},
                                                                                                                         'type': 'array'},
                                                                                                                'dtype': {'minLength': 1,
                                                                                                                          'type': 'string'},
                                                                                                                'name': {'minLength': 1,
                                                                                                                         'type': 'string'}},
                                                                                                 'required': ['dtype',
                                                                                                              'dims',
                                                                                                              'name'],
                                                                                                 'type': 'object'},
                                                                        'type': 'object'},
                                                          'dims': {'additionalProperties': {'minimum': 0,
                                                                                            'type': 'integer'},
                                                                   'type': 'object'},
                                                          'open_params_schema': {'additionalProperties': True,
                                                                                 'type': 'object'},
                                                          'spatial_res': {'exclusiveMinimum': 0.0,
                                                                          'type': 'number'},
                                                          'time_period': {'minLength': 1,
                                                                          'type': 'string'},
                                                          'time_range': {'items': [{'format': 'date',
                                                                                    'type': ['string',
                                                                                             'null']},
                                                                                   {'format': 'date',
                                                                                    'type': ['string',
                                                                                             'null']}],
                                                                         'type': ['array',
                                                                                  'null']}},
                                           'required': ['data_id',
                                                        'data_type'],
                                           'type': 'object'},
                    'size_estimation': {'additionalProperties': True,
                                        'type': 'object'}},
     'required': ['dataset_descriptor',
                  'size_estimation',
                  'cost_estimation'],
     'type': 'object'}

On instance:
    {'output': ['Cube generator configuration loaded from '
                '/data/user-code/a414f0ac6070db124b3999f70cc4790c5-38599831-c96d-438f.yaml.',
                'Result written to '
                '/data/results/a414f0ac6070db124b3999f70cc4790c5-38599831-c96d-438f.json'],
     'result': {'cost_estimation': {'available': 1000000,
                                    'limit': 10000,
                                    'required': 192},
                'dataset_descriptor': {'bbox': [7.0, 53.0, 9.0, 55.0],
                                       'crs': 'WGS84',
                                       'data_id': 'S2L2A_B04_B05_7_53_9_55.zarr',
                                       'data_type': 'dataset',
                                       'data_vars': {'B04': {'dims': ['time',
                                                                      'lat',
                                                                      'lon'],
                                                             'dtype': 'float32',
                                                             'name': 'B04'},
                                                     'B05': {'dims': ['time',
                                                                      'lat',
                                                                      'lon'],
                                                             'dtype': 'float32',
                                                             'name': 'B05'}},
                                       'dims': {'lat': 2048,
                                                'lon': 2048,
                                                'time': 6},
                                       'spatial_res': 0.001,
                                       'time_period': '1D',
                                       'time_range': ['2019-08-05',
                                                      '2019-08-10']},
                'size_estimation': {'image_size': [2048, 2048],
                                    'num_bytes': 201326592,
                                    'num_requests': 48,
                                    'num_tiles': [2, 2],
                                    'num_variables': 2,
                                    'tile_size': [1024, 1024]}},
     'status': 'ok'}

The above exception was the direct cause of the following exception:

RuntimeError                              Traceback (most recent call last)
/tmp/ipykernel_12096/1451062372.py in <module>
----> 1 result = generator.get_cube_info(request)
      2 result

~/Desktop/projects/xcube-dev/xcube/core/gen2/remote/generator.py in get_cube_info(self, request)
    117                                  headers=self.auth_headers)
    118         response_type = CubeInfoWithCosts if self._access_token else CubeInfo
--> 119         cube_info = self._parse_response(response,
    120                                          response_type=response_type,
    121                                          request_data=request_data)

~/Desktop/projects/xcube-dev/xcube/core/gen2/remote/generator.py in _parse_response(self, response, response_type, request_data)
    240             return response_type.from_dict(response_data)
    241         except Exception as e:
--> 242             raise RuntimeError(f'internal error: unexpected response'
    243                                f' from API call {response.url}: {e}') from e
    244 

RuntimeError: internal error: unexpected response from API call https://dev.xcube-gen.brockmann-consult.de/api/v2/cubegens/info: 'dataset_descriptor' is a required property

Failed validating 'required' in schema:
    {'additionalProperties': True,
     'properties': {'cost_estimation': {'additionalProperties': False,
                                        'properties': {'available': {'minimum': 0,
                                                                     'type': 'integer'},
                                                       'limit': {'minimum': 0,
                                                                 'type': 'integer'},
                                                       'required': {'minimum': 0,
                                                                    'type': 'integer'}},
                                        'required': ['required'],
                                        'type': 'object'},
                    'dataset_descriptor': {'additionalProperties': False,
                                           'properties': {'attrs': {'additionalProperties': True,
                                                                    'type': 'object'},
                                                          'bbox': {'items': [{'type': 'number'},
                                                                             {'type': 'number'},
                                                                             {'type': 'number'},
                                                                             {'type': 'number'}],
                                                                   'type': 'array'},
                                                          'coords': {'additionalProperties': {'additionalProperties': False,
                                                                                              'properties': {'attrs': {'additionalProperties': True,
                                                                                                                       'type': 'object'},
                                                                                                             'chunks': {'items': {'minimum': 0,
                                                                                                                                  'type': 'integer'},
                                                                                                                        'type': 'array'},
                                                                                                             'dims': {'items': {'minLength': 1,
                                                                                                                                'type': 'string'},
                                                                                                                      'type': 'array'},
                                                                                                             'dtype': {'minLength': 1,
                                                                                                                       'type': 'string'},
                                                                                                             'name': {'minLength': 1,
                                                                                                                      'type': 'string'}},
                                                                                              'required': ['dtype',
                                                                                                           'dims',
                                                                                                           'name'],
                                                                                              'type': 'object'},
                                                                     'type': 'object'},
                                                          'crs': {'minLength': 1,
                                                                  'type': 'string'},
                                                          'data_id': {'minLength': 1,
                                                                      'type': 'string'},
                                                          'data_type': {'minLength': 1,
                                                                        'type': 'string'},
                                                          'data_vars': {'additionalProperties': {'additionalProperties': False,
                                                                                                 'properties': {'attrs': {'additionalProperties': True,
                                                                                                                          'type': 'object'},
                                                                                                                'chunks': {'items': {'minimum': 0,
                                                                                                                                     'type': 'integer'},
                                                                                                                           'type': 'array'},
                                                                                                                'dims': {'items': {'minLength': 1,
                                                                                                                                   'type': 'string'},
                                                                                                                         'type': 'array'},
                                                                                                                'dtype': {'minLength': 1,
                                                                                                                          'type': 'string'},
                                                                                                                'name': {'minLength': 1,
                                                                                                                         'type': 'string'}},
                                                                                                 'required': ['dtype',
                                                                                                              'dims',
                                                                                                              'name'],
                                                                                                 'type': 'object'},
                                                                        'type': 'object'},
                                                          'dims': {'additionalProperties': {'minimum': 0,
                                                                                            'type': 'integer'},
                                                                   'type': 'object'},
                                                          'open_params_schema': {'additionalProperties': True,
                                                                                 'type': 'object'},
                                                          'spatial_res': {'exclusiveMinimum': 0.0,
                                                                          'type': 'number'},
                                                          'time_period': {'minLength': 1,
                                                                          'type': 'string'},
                                                          'time_range': {'items': [{'format': 'date',
                                                                                    'type': ['string',
                                                                                             'null']},
                                                                                   {'format': 'date',
                                                                                    'type': ['string',
                                                                                             'null']}],
                                                                         'type': ['array',
                                                                                  'null']}},
                                           'required': ['data_id',
                                                        'data_type'],
                                           'type': 'object'},
                    'size_estimation': {'additionalProperties': True,
                                        'type': 'object'}},
     'required': ['dataset_descriptor',
                  'size_estimation',
                  'cost_estimation'],
     'type': 'object'}

On instance:
    {'output': ['Cube generator configuration loaded from '
                '/data/user-code/a414f0ac6070db124b3999f70cc4790c5-38599831-c96d-438f.yaml.',
                'Result written to '
                '/data/results/a414f0ac6070db124b3999f70cc4790c5-38599831-c96d-438f.json'],
     'result': {'cost_estimation': {'available': 1000000,
                                    'limit': 10000,
                                    'required': 192},
                'dataset_descriptor': {'bbox': [7.0, 53.0, 9.0, 55.0],
                                       'crs': 'WGS84',
                                       'data_id': 'S2L2A_B04_B05_7_53_9_55.zarr',
                                       'data_type': 'dataset',
                                       'data_vars': {'B04': {'dims': ['time',
                                                                      'lat',
                                                                      'lon'],
                                                             'dtype': 'float32',
                                                             'name': 'B04'},
                                                     'B05': {'dims': ['time',
                                                                      'lat',
                                                                      'lon'],
                                                             'dtype': 'float32',
                                                             'name': 'B05'}},
                                       'dims': {'lat': 2048,
                                                'lon': 2048,
                                                'time': 6},
                                       'spatial_res': 0.001,
                                       'time_period': '1D',
                                       'time_range': ['2019-08-05',
                                                      '2019-08-10']},
                'size_estimation': {'image_size': [2048, 2048],
                                    'num_bytes': 201326592,
                                    'num_requests': 48,
                                    'num_tiles': [2, 2],
                                    'num_variables': 2,
                                    'tile_size': [1024, 1024]}},
     'status': 'ok'}

**Pull request merged**